tests/test_mask_helper.py

```diff
@@ -0,0 +1,137 @@
+    print(hm)
+    HyperParameterMasking.check_instantiation(hm)
+    HyperParameterMasking.check_matching(hm, hm['hyps'], hm['cutoffs'])
+
+def test_initialization():
+    pm = HyperParameterMasking(species=['O', 'C', 'H'],
+                          bonds=[['*', '*'], ['O','O']],
+                          triplets=[['*', '*', '*'], ['O','O', 'O']],
+                          parameters={'bond0':[1, 0.5], 'bond1':[2, 0.2],
+                                'triplet0':[1, 0.5], 'triplet1':[2, 0.2],
+                                'cutoff2b':2, 'cutoff3b':1})
+    hm = pm.hyps_mask
+    print(hm)
+    HyperParameterMasking.check_instantiation(hm)
+    HyperParameterMasking.check_matching(hm, hm['hyps'], hm['cutoffs'])
+
+def test_opt():
+    pm = HyperParameterMasking(species=['O', 'C', 'H'],
+                          bonds=[['*', '*'], ['O','O']],
+                          triplets=[['*', '*', '*'], ['O','O', 'O']],
+                          parameters={'bond0':[1, 0.5, 1], 'bond1':[2, 0.2, 2],
+                                'triplet0':[1, 0.5], 'triplet1':[2, 0.2],
+                                'cutoff2b':2, 'cutoff3b':1},
+                          constraints={'bond0':[False, True]})
+    hm = pm.hyps_mask
+    print(hm)
+    HyperParameterMasking.check_instantiation(hm)
+    HyperParameterMasking.check_matching(hm, hm['hyps'], hm['cutoffs'])
+
+def test_randomization():
+    pm = HyperParameterMasking(species=['O', 'C', 'H'],
+                          bonds=True, triplets=True,
+                          mb=False, allseparate=True,
+                          random=True,
+                          parameters={'cutoff2b': 7,
+                              'cutoff3b': 4.5,
+                              'cutoffmb': 3},
+                          verbose=True)
+    hm = pm.hyps_mask
+    print(hm)
+    HyperParameterMasking.check_instantiation(hm)
+    HyperParameterMasking.check_matching(hm, hm['hyps'], hm['cutoffs'])
+    name = pm.find_group('specie', 'O')
+    print("find group name for O", name)
+    name = pm.find_group('bond', ['O', 'C'])
+    print("find group name for O-C", name)
+
+def test_from_dict():
+    pm = HyperParameterMasking(species=['O', 'C', 'H'],
+                          bonds=True, triplets=True,
+                          mb=False, allseparate=True,
+                          random=True,
+                          parameters={'cutoff2b': 7,
+                              'cutoff3b': 4.5,
+                              'cutoffmb': 3},
+                          verbose=True)
+    hm = pm.hyps_mask
+    HyperParameterMasking.check_instantiation(hm)
+    HyperParameterMasking.check_matching(hm, hm['hyps'], hm['cutoffs'])
+    print(hm['hyps'])
+    print("obtain test hm", hm)
+
+    pm1 = HyperParameterMasking.from_dict(hm, verbose=True)
+    print("from_dict")
+    hm1 = pm1.generate_dict()
+    print(hm['hyps'])
+    print(hm1['hyps'][:33], hm1['hyps'][33:])
+
+    dumpcompare(hm, hm1)
```

Comment:
It occurs to me that it might also make sense to try up a few wrong or incorrect setups, and with an exception catch, also test that check_instantiation correctly vetoes them. To unit test the tests in a way...